### PR TITLE
feat(routes): make topmost router pivot over WalletInfos

### PR DESCRIPTION
**Pull request by mariocao**
# PR Prelude

Thank you for contributing to sheikah! :)

**Please fill in this template (by putting an `x` inside
the brackets) _before_ filing your PR:**

- [x] I have read and understood [CODE_OF_CONDUCT][code] document.
- [x] I have read and understood [CONTRIBUTING][cont] document.
- [x] I have read and understood [STYLEGUIDE][style] document.
- [x] I have included tests for the changes in my PR. If not, I have included a rationale for why I haven't.
- [x] My code is formatted and is accepted by `yarn fmt`.
- [x] **I understand my PR may be closed if it becomes obvious I didn't
  actually perform all of these steps.**

# Why this change is necessary and useful

The router includes the following routes:
+ `/main` renders `MainPage`
+ `forms/login` renders `LoginFormContainer`
+ `forms/wallet` renders `SignUpPage`
+ `/` redirects to the login or sign-up form depending on the existence of wallet infos in the redux store

PS: some route paths have been hardcoded because they have not yet been exported from their corresponding containers. Nevertheless, they have been tagged with `TODO` comments.

[code]: https://github.com/witnet/sheikah/blob/master/.github/CODE_OF_CONDUCT.md
[cont]: https://github.com/witnet/sheikah/blob/master/.github/CONTRIBUTING.md
[style]: https://github.com/witnet/sheikah/blob/master/docs/STYLEGUIDE.md

### CODE REVIEW

File: app/renderer/routes.tsx [hunk 1]
```diff
@@ -1,41 +1,71 @@
...
-import { MainPage } from "app/renderer/ui/pages/main"
 import { StoreState } from "app/renderer/store"
-
-import { RedirectedRoute } from "app/renderer/utils/redirectedRoute"
-import { ifWallet, ifWallets } from "app/renderer/utils/guards"
-import LoginFormContainer from "app/renderer/ui/containers/LoginForm"
+import App from "app/renderer/ui/containers/App"
+import {
+  default as LoginFormContainer,
+  PATHS as PATHS_LOGIN_FORM
```
Comment (aesedepece) [inline app/renderer/routes.tsx]: `LOGIN_FORM_PATHS`
Sorry for my "pickiness" :rofl:

File: app/renderer/routes.tsx [hunk 2]
```diff
@@ -1,41 +1,71 @@
...
+          // TODO: update with PATHS from new wallet form container
+          path="/forms/wallet"
+          render={newWalletRenderer}
+        />
+        <Route
+          // TODO: update with PATHS from Main container
+          path="/main"
+          render={mainRenderer}
+        />
+        <Route
```
Comment (aesedepece) [inline app/renderer/routes.tsx]: Can we move this upwards so the two forms lay together?

File: app/renderer/ui/containers/LoginForm.tsx [hunk 3]
```diff
@@ -17,8 +17,9 @@ import WalletPasswordRequest from "app/renderer/ui/components/loginForm/steps/wa
  * Paths of the routes that are used in this container
  */
 export const PATHS = {
-  WALLET_SELECTION: "/login",
-  WALLET_PASSWORD_REQUEST: "/login/walletPasswordRequest",
+  ROOT: "/forms/login",
+  WALLET_SELECTION: "/forms/login/welcome",
+  WALLET_PASSWORD_REQUEST: "/forms/login/password",
```
Comment (aesedepece) [inline app/renderer/ui/containers/LoginForm.tsx]: I kind of prefer `WALLET_PASSWORD_ENTRY` or `WALLET_PASSWORD_PROMPT`

File: app/renderer/routes.tsx [hunk 4]
```diff
@@ -1,41 +1,71 @@
...
+          path={PATHS_LOGIN_FORM.ROOT}
```
Comment (aesedepece) [inline app/renderer/routes.tsx]: This forces the router inside the login form to define a default route pointing to its own welcome screen. As we discussed, we could drop default routes and make all routes explicit altogether.

File: app/renderer/routes.tsx (hunk 2 above)
Comment (mmartinbar) [reply app/renderer/routes.tsx]: Sure, thanks!

**Pull request merged**